### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by drupol**
We need to update the module documentation with code examples and also, the list of Drupal 8 files that we've copied over and that has been altered.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 
-Service Container is an API module based on ctools to enable a Drupal 7 quick and easy lightweight service container with 100% unit test coverage.
+Service Container is an API module based on [ctools](https://www.drupal.org/project/ctools) to enable a Drupal 7 quick and easy lightweight service container with 100% unit test coverage.
 
 It is very similar in syntax to a Symfony container, but was written from scratch as a symfony dependency was not wanted - using some of Drupal 8 Core and Component directly. They will likely depend on a drupal8core project in the future - but for now the copy is fine.
 
 This allows to use an extensible service container (like in Drupal 8) and write modules in Drupal 7 as if they were using Drupal 8.
 
-The main benefit is being able to use unit testing.
+The main benefit is being able to use unit testing but also to write Drupal 7 module with Drupal 8 style of coding in mind.
```
Comment (dawehner) [inline README.md:9]: +1 that is indeed an important point!

File: README.md [hunk 2]
```diff
@@ -62,21 +62,28 @@ $plugin = array(
...
 * database ('database')
 * key value store ('keyvalue', 'keyvalue.database')
-* a wrapper for variable_get()  / variable_set() ('variable')
+* variable, a wrapper for variable_get() / variable_set()
 * A lock ('lock')
 * A wrapper for url() / l() ('url_generator', 'link_generator')
-* TODO
+* Flood, a wrapper for the flood mechanisms
+* Messenger, a wrapper for displaying messages
+* Drupal 7 Legacy, a wrapper for accessing the Drupal 7 legacy functions.
```
Comment (dawehner) [inline README.md:79]: Awesome work!

**Comment by drupol** [comment]
Thanks ! Now we have to make a list of files that has been altered. Should I open another PR ?

**Comment by fabianx-ai** [comment]
Yes, please.

**Comment by drupol** [comment]
Done here: https://github.com/LionsAd/service_container/pull/32

**Pull request merged**